file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemProvider.kt

```diff
@@ -370,14 +378,21 @@ internal class AttributeCachingFileSystemProvider : FileSystemProvider() {
             "dos:*"
         } else if (attribute.startsWith("posix")) {
             "posix:*"
+        } else if (attribute.startsWith("acl")) {
+            "acl:*"
         } else {
             "*"
         }
 
         // Even if we have a single attribute only we should get the entire attribute class for that single
         // attribute to properly set the cache.
-        val attributesObject = getAttributesClassFromPathProvider(path, attributeClassName)
-        path.setAttributeByName(attributeClassName, attributesObject)
+        val attributesView: FileAttributeView? = getAttributesClassFromPathProvider(path, attributeClassName)
+        if (attributeClassName == "acl:*") {
+            path.aclFileAttributeView = attributesView as? AclFileAttributeView
+        } else {
+            val basicFileAttributesView = attributesView as? BasicFileAttributeView
+            path.setAttributeByName(attributeClassName, basicFileAttributesView?.readAttributes())
+        }
```

Comment:
`AclFileAttributeView` is a `FileAttributeView`, so I think handling acl in `setAttributeByName` is perfectly appropriate and simplifies the logic.

Reply:
SetAttributeByName takes in a BasicFileAttributes object, not the view, so we cannot do this

Reply:
okay i see what you mean this makes sense and id just edit what happens within setAttributeViewByName